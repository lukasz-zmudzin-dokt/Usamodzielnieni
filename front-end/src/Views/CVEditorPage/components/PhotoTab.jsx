import React from "react";
import { Form, Col } from "react-bootstrap";
import { CVEditorTab } from ".";
import movie_1 from "assets/movie_1.png";

class PhotoTab extends React.Component {
  constructor(props) {
    super(props);
    this.fileInput = React.createRef();
  }

  onChange = e => {
    this.props.onChange(this.fileInput.files[0]);
    console.log(this.fileInput.files[0]);
  };

  render() {
    return (
      <CVEditorTab
        title="Zdjęcie"
        movie={movie_1}
        onPrevClick={this.props.onPrevClick}
        disabled={this.props.disabled}
      >
        <Form.Group as={Col} controlId="photo">
          <Form.Label>Wybierz zdjęcie:</Form.Label>
          <Form.Control
            name="photo"
            type="file"
            ref={ref => (this.fileInput = ref)}
            onChange={this.onChange}
            accept="image/*"
          />
        </Form.Group>
      </CVEditorTab>
    );
  }
}

export default PhotoTab;
