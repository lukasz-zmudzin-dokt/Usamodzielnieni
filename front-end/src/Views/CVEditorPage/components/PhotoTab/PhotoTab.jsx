import React from "react";
import { Form } from "react-bootstrap";
import { CVEditorTab } from "..";
import movie_1 from "assets/movie_1.png";
import proxy from "config/api";
import { UserContext } from "context";
import { approveFileSize } from "utils/approveFile/approveFile";

class PhotoTab extends React.Component {
  constructor(props) {
    super(props);
    this.fileInput = React.createRef();
  }

  onChange = (e) => {
    const file = this.fileInput.files[0];
    if (approveFileSize(file) === true) {
      this.props.onChange(this.fileInput.files[0]);
    } else {
      this.props.alertContext.showAlert(
        "Wybrany plik jest za duży. Maksymalny rozmiar pliku to 15 MB."
      );
      this.fileInput = React.createRef();
    }
  };

  setLabel = () => {
    const label =
      this.props.data !== null
        ? this.props.data.name
        : this.props.hasPhoto
        ? "Poprzednie zdjęcie"
        : "Wybierz zdjęcie";
    return label;
  };

  render() {
    return (
      <CVEditorTab
        title="Zdjęcie"
        movie={movie_1}
        onPrevClick={this.props.onPrevClick}
        onSubmit={this.props.onSubmit}
        comments={this.props.comments}
        loading={this.props.loading}
        error={this.props.error}
        showComments={this.props.showComments}
        disabled={this.props.disabled}
        isNew={this.props.isNew}
        group_type={this.context.data.group_type}
        video={this.props.video}
        errVid={this.props.errVid}
        formTab={this.props.formTab}
      >
        <Form>
          <Form.Group>
            <Form.Label htmlFor="custom-file">Zdjęcie:</Form.Label>
            <Form.File
              id="custom-file"
              custom
              label={this.setLabel()}
              ref={(ref) => (this.fileInput = ref)}
              onChange={this.onChange}
              accept="image/*"
              data-browse="Dodaj"
              // value="this.props.data"
            />
          </Form.Group>
          {this.props.templateList?.length > 0 && (
            <Form.Group>
              <Form.Label>
                Kolor CV: (
                <a
                  target="_blank"
                  href={proxy.plain + "/static/" + this.props.template + ".pdf"}
                  rel="noopener noreferrer"
                >
                  Kliknij <u>tutaj</u> żeby zobaczyć przykładowe CV w tym
                  kolorze
                </a>
                )
              </Form.Label>
              <Form.Control
                as="select"
                id="template"
                value={this.props.template}
                onChange={(e) => this.props.setTemplate(e.target.value)}
              >
                {this.props.templateList.map((item) => (
                  <option
                    key={item}
                    value={item}
                    style={{ backgroundColor: item }}
                  >
                    {item}
                  </option>
                ))}
              </Form.Control>
            </Form.Group>
          )}
        </Form>
      </CVEditorTab>
    );
  }
}

PhotoTab.contextType = UserContext;

export default PhotoTab;
