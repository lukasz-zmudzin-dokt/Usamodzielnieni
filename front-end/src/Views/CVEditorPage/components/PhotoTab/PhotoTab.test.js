import React from "react";
import { render, fireEvent } from "@testing-library/react";
import PhotoTab from "./PhotoTab";

describe("photo tab tests", () => {
  let props = {};
  //let data;
  beforeAll(() => {
    props = {
      data: {},
      onChange: (data) => {},
      onPrevClick: () => {},
      comments: "",
    };
  });

  it("should render without crashing", () => {
    const { container } = render(<PhotoTab></PhotoTab>);
    expect(container).toMatchSnapshot();
  });
  it("should react on input change", () => {
    const { getByLabelText } = render(<PhotoTab {...props}></PhotoTab>);
    const input = getByLabelText("Wybierz zdjÄ™cie:");
    fireEvent.change(input, {
      target: { files: ["movie_4.png"] },
    });
    expect(input.files[0]).toBe("movie_4.png");
  });
});
