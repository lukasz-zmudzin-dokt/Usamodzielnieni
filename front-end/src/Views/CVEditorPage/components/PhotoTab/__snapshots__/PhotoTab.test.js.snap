// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`photo tab tests should render without crashing 1`] = `
<div>
  <div>
    <h3>
      Zdjęcie
    </h3>
    <div
      class="form-group"
    >
      <img
        alt=""
        class="CVEditorTab__img"
        src="movie_1.png"
      />
    </div>
    <div
      class="form-group col"
    >
      <label
        class="form-label"
        for="photo"
      >
        Wybierz zdjęcie:
      </label>
      <input
        accept="image/*"
        class="form-control-file"
        id="photo"
        name="photo"
        type="file"
      />
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6 col-12"
      >
        <button
          class="form_navigation_prev btn btn-primary btn-block"
          disabled=""
          type="button"
        >
          ← Wstecz
        </button>
      </div>
      <div
        class="col-md-6 col-12"
      >
        <button
          class="btn btn-success btn-block"
          id="saveButton"
          type="submit"
        >
          Generuj CV
        </button>
      </div>
    </div>
  </div>
</div>
`;
